### PR TITLE
Add CMake option to disable labels

### PR DESCRIPTION
**Pull request by pkomon-tgm**
Labels introduced a dependency to mapbox/variant that uses the function `std::result_of`, which was removed in C++20. While some compilers don't care as long as it is not actually used, unfortunately MSVC does care.

This PR introduces a CMake option to disable labels and thus side step compile issues with MSVC until a long-term solution is implemented (i.e. the deprecated dependency is removed).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -27,14 +27,19 @@ option(ALP_ENABLE_TRACK_OBJECT_LIFECYCLE "enables debug cmd printout of construc
...
+option(ALP_ENABLE_LABELS "Enables label rendering" OFF)
 
 set(ALP_EXTERN_DIR "extern" CACHE STRING "name of the directory to store external libraries, fonts etc..")
 
 if(ALP_ENABLE_TRACK_OBJECT_LIFECYCLE)
     add_definitions(-DALP_ENABLE_TRACK_OBJECT_LIFECYCLE)
 endif()
 
+if(ALP_ENABLE_LABELS)
+    add_compile_definitions(ALP_ENABLE_LABELS)
```
Comment (adam-ce) [inline CMakeLists.txt:40]: better use target `target_compile_definitions` in the nucleus cmake file

File: CMakeLists.txt [hunk 2]
```diff
@@ -27,14 +27,19 @@ option(ALP_ENABLE_TRACK_OBJECT_LIFECYCLE "enables debug cmd printout of construc
 option(ALP_ENABLE_APP_SHUTDOWN_AFTER_60S "Shuts down the app after 60S, used for CI testing with asan." OFF)
 option(ALP_ENABLE_LTO "Enable link time optimisation." OFF)
 option(ALP_ENABLE_GL_ENGINE "Enable OpenGL/WebGL engine" ON)
-option(ALP_ENABLE_AVLANCHE_WARNING_LAYER "Enables avalanche warning layer (requires Qt Gui in nucleus)" ON)
+option(ALP_ENABLE_AVLANCHE_WARNING_LAYER "Enables avalanche warning layer (requires Qt Gui in nucleus)" OFF)
+option(ALP_ENABLE_LABELS "Enables label rendering" OFF)
```
Comment (adam-ce) [inline CMakeLists.txt:31]: please keep the labels on by default.

they are quite core to maps, and the std variant will be fixed soon

**Review by adam-ce — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (pkomon-tgm) [reply CMakeLists.txt:40]: We need it in multiple targets (`nucleus`, `gl_engine`), so should I add it to each individually?

Comment (pkomon-tgm) [reply CMakeLists.txt]: fixed

**Pull request merged**